beacon-chain/forkchoice/protoarray/optimistic_sync_test.go

```diff
@@ -535,15 +535,16 @@ func TestFindSyncedTip(t *testing.T) {
 	}
 	for _, tc := range tests {
 		f.store.nodesLock.RLock()
-		defer f.store.nodesLock.RUnlock()
 		node := f.store.nodes[f.store.nodesIndices[tc.root]]
 		syncedTips := &optimisticStore{
 			validatedTips: tc.tips,
 		}
 		syncedTips.RLock()
-		defer syncedTips.RUnlock()
 		idx, err := f.store.findSyncedTip(ctx, node, syncedTips)
 		require.NoError(t, err)
 		require.Equal(t, tc.wanted, f.store.nodes[idx].root)
+
+		f.store.nodesLock.RUnlock()
+		syncedTips.RUnlock()
```

Comment:
I personally find deferred unlocking coded next to locking more readable than this. I don't feel strongly about it though.

Reply:
In this case it's a bug (introduced by me, nice catch @terencechain) the deferral happens when the loop ends, not within each round.

Reply:
The description doesn't mention anything about a bug fix though. I wonder if it is a deliberate fix.

Reply:
We typically don't do defer in a loop because of a potential memory leak, but this is ok because it's a test. I fixed it anyway because my code inspector complained. (i.e. clean up warnings)